uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Service/CiviHRStatsGatherer.php

```diff
@@ -0,0 +1,172 @@
+  }
+
+  /**
+   * Sets the site base URL
+   *
+   * @param CiviHRStatistics $stats
+   */
+  private function setBaseUrl(CiviHRStatistics $stats) {
+    $config = & CRM_Core_Config::singleton();
+    $stats->setSiteUrl($config->userFrameworkBaseURL);
+  }
+
+  /**
+   * Fetches counts and sets them for all required entities
+   *
+   * @param CiviHRStatistics $stats
+   * @throws CiviCRM_API3_Exception
+   */
+  private function setEntityCounts(CiviHRStatistics $stats) {
+    $taskAssignmentsKey = 'uk.co.compucorp.civicrm.tasksassignments';
+    $leaveAndAbsenceKey = 'uk.co.compucorp.civicrm.hrleaveandabsences';
+    $recruitmentKey = 'org.civicrm.hrrecruitment';
+
+    $stats->setEntityCount('contact', $this->getEntityCount('Contact'));
+
+    if (ExtensionHelper::isExtensionEnabled($taskAssignmentsKey)) {
+      $stats->setEntityCount('assignment', $this->getEntityCount('Assignment'));
+      $stats->setEntityCount('task', $this->getEntityCount('Task'));
+      $stats->setEntityCount('document', $this->getEntityCount('Document'));
+    }
+
+    if (ExtensionHelper::isExtensionEnabled($leaveAndAbsenceKey)) {
+      // leave requests in last 100 days
+      $format = 'Y-m-d H:i:s';
+      $oneHundredDaysAgo = (new \DateTime('today - 100 days'))->format($format);
+      $params = ['from_date' => ['>=' => $oneHundredDaysAgo]];
+      $last100DaysCount = (int) civicrm_api3('LeaveRequest', 'getcount', $params);
+      $stats->setEntityCount('leaveRequestInLast100Days', $last100DaysCount);
+
+      // total leave requests
+      $leaveRequestCount = $this->getEntityCount('LeaveRequest');
+      $stats->setEntityCount('leaveRequest', $leaveRequestCount);
+    }
+
+    if (ExtensionHelper::isExtensionEnabled($recruitmentKey)) {
+      $stats->setEntityCount('vacancy', $this->getEntityCount('HRVacancy'));
+    }
+
+    $stats->setEntityCount('drupalUser', $this->getEntityCount('UFMatch'));
+  }
+
+  /**
+   * Gets the number of a certain entity in the system.
+   *
+   * @param string $entity
+   *
+   * @return int
+   */
+  private function getEntityCount($entity) {
+    return (int) civicrm_api3($entity, 'getcount');
+  }
+
+  /**
+   * Fetches contact subtypes
+   *
+   * @param CiviHRStatistics $stats
+   * @throws CiviCRM_API3_Exception
+   */
+  private function setContactSubtypes(CiviHRStatistics $stats) {
+    $params = ['parent_id' => ['IS NULL' => 1]];
+    $contactTypes = civicrm_api3('ContactType', 'get', $params)['values'];
+    foreach ($contactTypes as $contactType) {
+      $name = $contactType['name'];
+      $count = civicrm_api3('Contact', 'getcount', ['contact_type' => $name]);
+      $stats->setContactSubtypeCount($name, (int) $count);
+    }
+  }
+
+  /**
+   * Fetches report configurations
+   *
+   * @param CiviHRStatistics $stats
+   */
+  private function setReportConfigurations(CiviHRStatistics $stats) {
+    // Reports are only available in Drupal
+    if (!$this->cmsVariableService instanceof DrupalVariableService) {
```

Comment:
Yeah, I agree. The only reason I did it this way was to avoid more static calls, and instead just trying to determine whether it was Drupal based on existing depenencies. I can change it to use the same check from the factory class, which is clearer.